### PR TITLE
fix(web): color japanese public holidays like sundays in the calendar

### PR DESCRIPTION
**Pull request by kurone-kito**
## Summary

Colors Japanese public holidays like Sundays in the schedule calendar
(color only — no holiday-name text or legend entry, per the operator's
explicit decision recorded on the issue).

- `packages/fetcher`: adds `@holiday-jp/holiday_jp` and a new
  `holiday.mts` helper that determines the JST calendar date from an
  epoch (via `Intl` with `timeZone: 'Asia/Tokyo'`, then a fresh local
  `Date` from those parts) before checking `holiday_jp.isHoliday`,
  since that library reads back the `Date`'s *local* getters and would
  answer incorrectly under a non-JST host process (this repo's
  `push.yml` CI runs UTC). `EventDetailRow` gains an optional
  `holiday?: true` field (omitted, not `false`, on non-holidays, to
  keep `data.json` minimal), set by `toRowMapper`.
- `packages/web`: `Row.tsx` emits a `holiday` class on the date cell
  *instead of* the week class when set — decided in class-emission
  logic, not CSS cascade order, so a holiday Saturday cannot also
  receive `.sat` styling. `Table.tsx` styles `.holiday` the same as
  `.sun`. `Row.stories.ts` gets a `holiday` toggle control.
- Tests: `holiday.test.mts` and `toRow.test.mts` cover the JST-vs-UTC
  boundary case (an epoch that's a holiday Saturday in JST but the
  previous day/weekday in UTC), and `Row.test.tsx` covers the
  class-emission behavior with `@solidjs/testing-library`.

`detectChange.mts` (the #130 display projection) needs no change — it
serializes `toRowMapper`'s output directly, so `holiday` flows through
automatically.

### Included infra fix (own commit)

Root's `"test"` script called `vitest run` directly, which orchestrates
`vitest.workspace.json` using root's own stale `vitest@2.1.9`
devDependency — independent of each package's own pin
(`packages/web` pins `^3.2.6`). This repo's first real Solid-rendering
test (`Row.test.tsx`) exposed that the version-2 workspace orchestrator
resolves `solid-js` to its **server** build instead of the DOM build,
crashing with "Client-only API called on the server side" — even though
the identical test passes when vitest runs from inside `packages/web`
directly. Changed `"test": "vitest run"` to `"test": "pnpm run -r
test"`, matching the existing `"build": "pnpm run -r build"`
convention. Verified green locally under both `TZ=UTC` and
`TZ=Asia/Tokyo`. This is a latent, pre-existing gap (no prior web test
exercised real component rendering) required to land this issue's
component tests, and would have hit #164 identically. Left the now-
unused root `vitest` devDependency and `vitest.workspace.json` in place
(still usable for IDE test-runner integration) — removing them is a
smaller, separate cleanup out of this issue's scope.

### Verification

- `pnpm run lint && pnpm run test` green, both `TZ=UTC` and
  `TZ=Asia/Tokyo`.
- Full local production build succeeds (real `.env` credentials). This
  week's real calendar data has no holidays, so I additionally
  substituted a fixture `data.json` with a holiday Saturday and
  rebuilt just `vinxi build`: the prerendered HTML emitted
  `class="date holiday"` (not `sat`), and the compiled CSS carried
  `.holiday{background-color:#ef444499!important}` — confirming
  holiday-wins-over-Saturday end to end.
- Storybook's dev server is broken in this environment (pre-existing,
  unrelated); the fixture-build check above substitutes for a manual
  Storybook look at the new `holiday` control.

## Test plan

- [x] `pnpm run lint` passes
- [x] `pnpm run test` passes under `TZ=UTC`
- [x] `pnpm run test` passes under `TZ=Asia/Tokyo`
- [x] Full production build succeeds
- [x] Fixture-data build confirms `.holiday` class + red styling wins
      over `.sat` on a holiday Saturday
- [ ] CI (build/lint/test on `issue/*`) green

Closes #134

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/fetcher/src/holiday.mts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * process runs under.
+ * @param epoch The epoch time, in milliseconds.
+ * @returns The date whose local calendar date is the JST calendar date.
+ */
+export const toJstCalendarDate = (epoch: number): Date => {
+  const parts = jstDateFormatter.formatToParts(new Date(epoch));
+  const get = (type: 'day' | 'month' | 'year'): number =>
+    Number(parts.find((part) => part.type === type)?.value);
+  return new Date(get('year'), get('month') - 1, get('day'));
+};
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply packages/fetcher/src/holiday.mts:39]: **Accepted** — implemented as suggested in e5255d2: `toJstCalendarDate` now throws a descriptive error if `Intl.DateTimeFormat#formatToParts` ever omits a requested part, instead of silently coercing to `NaN`/`Invalid Date`. Covered by a new test that stubs `formatToParts` to exercise the guard.

File: packages/web/src/components/atoms/calendar/Row.test.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Render a {@link Row} and return its date cell.
+ * @param props The properties to pass to {@link Row}.
+ * @returns The date cell element, or `null` when not rendered.
+ */
+const renderDateCell = (
+  props: Parameters<typeof Row>[0],
+): HTMLElement | null => {
+  const { container } = render(() => <Row {...props} />);
+  return container.querySelector('.date');
+};
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply packages/web/src/components/atoms/calendar/Row.test.tsx]: **Accepted** — implemented as suggested in e5255d2: `renderDateCell` now asserts the queried `.date` cell is non-null and returns `HTMLElement` (not `HTMLElement | null`), so a missing date cell fails with a clear error at the helper instead of an opaque matcher error at each call site.
